src/types/redis_string.h

```diff
@@ -34,6 +34,8 @@ struct StringPair {
   Slice value;
 };
 
+enum class DelExOption { NONE, IFDEQ, IFDNE, IFEQ, IFNE };
```

Comment:
Maybe like this? We can also put the value inside it.

```suggestion
struct DelExOption {
  enum Type { NONE, IFDEQ, IFDNE, IFEQ, IFNE };
  Type type;
  std::string value;
};
```